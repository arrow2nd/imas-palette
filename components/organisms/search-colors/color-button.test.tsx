import { act, fireEvent, render } from '@testing-library/react'

import ColorButton from './color-button'

describe('ColorButtton', () => {
  const color = {
    name: '赤',
    hex: '#ff0000'
  }

  test('選択状態の表示が正しいか', () => {
    const props = {
      color: color,
      onClick: jest.fn()
    }

    const { container, rerender } = render(
      <ColorButton {...props} isSelected={false} />
    )
    const prevInnerHtml = container.innerHTML

    rerender(<ColorButton {...props} isSelected={true} />)

    expect(prevInnerHtml).not.toBe(container.innerHTML)
  })

  test('色が指定されていない時の表示は正しいか', () => {
    const { container } = render(
      <ColorButton
        color={{ name: '', hex: '' }}
        isSelected={false}
        onClick={jest.fn()}
      />
    )

    expect(container).toMatchSnapshot()
  })

  test('クリック時にコールバックが呼ばれるか', () => {
    const mock = jest.fn()
    const { getByTestId } = render(
      <ColorButton
        color={{ name: '', hex: '' }}
        isSelected={false}
        onClick={mock}
        dataTestId="search-color-button"
      />
    )

    act(() => {
      fireEvent.click(getByTestId('search-color-button'))
    })

    expect(mock).toBeCalled()
  })
})
