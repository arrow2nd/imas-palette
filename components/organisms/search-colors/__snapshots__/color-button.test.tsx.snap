// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorButton 色が指定されていない時の表示は正しいか 1`] = `
<div>
  <button
    class="mx-1 p-0 rounded-full border-2 border-transparent"
  >
    <div
      class="w-8 h-8 rounded-full border-2 bg-linear-to-r from-red-500 to-purple-500"
      title=""
    />
  </button>
</div>
`;
